### PR TITLE
Pointer events to touch events mapping

### PR DESCRIPTION
**Pull request by Warus15**
## Description

This PR adds mapping of pointer events to touch events. This change allows us to send touch events with pointers such as mouse or stylus, therefore making `ManualGestureHandler` usable with these pointers

## Test plan

Tested on example app

### CODE REVIEW

File: src/web/handlers/GestureHandler.ts [hunk 1]
```diff
@@ -394,62 +397,93 @@ export default abstract class GestureHandler {
...
-        event.allTouches[i].identifier
-      );
+    const trackerData = this.tracker.getData();
 
-      // When we simultaneously add 2 (or more) pointers to handler, both allTouches and changedTouches contains information
-      // about all pointers. Because first pointer tries to send event before second was added to tracker, mapped id of second pointer will be NaN.
-      // To avoid crashes, we check whether current id is NaN and, if that's the case, we return undfined and don't send event.
+    // This if handles edge case where all pointers have been cancelled
+    // When pointercancel is triggered, reset method is called. This means that tracker will be reset after first pointer being cancelled
+    // The problem is, that handler will receive another pointercancel event from rest oh the pointers
```
Comment (j-piasecki) [inline src/web/handlers/GestureHandler.ts:409]:
```suggestion
    // The problem is, that handler will receive another pointercancel event from the rest of the pointers
```

File: src/web/handlers/GestureHandler.ts [hunk 2]
```diff
@@ -394,62 +397,93 @@ export default abstract class GestureHandler {
...
-        absoluteX: event.allTouches[i].clientX,
-        absoluteY: event.allTouches[i].clientY,
+        x: element.lastX - rect.left,
+        y: element.lastY - rect.top,
+        absoluteX: element.lastX,
+        absoluteY: element.lastY,
+      });
+    });
+
+    // Each pointer send its own event, so we want changed touches to contain only the pointer that has changed.
```
Comment (j-piasecki) [inline src/web/handlers/GestureHandler.ts:427]:
```suggestion
    // Each pointer sends its own event, so we want changed touches to contain only the pointer that has changed.
```

File: src/web/tools/TouchEventManager.ts [hunk 3]
```diff
@@ -23,7 +23,9 @@ export default class TouchEventManager extends EventManager {
           !isPointerInBounds(this.view, {
             x: adaptedEvent.x,
             y: adaptedEvent.y,
-          })
+          }) ||
+          //@ts-ignore touchType field does exist
+          event.changedTouches[i].touchType === 'stylus'
```
Comment (j-piasecki) [inline src/web/tools/TouchEventManager.ts:28]: I'd comment why you're skipping stylus

File: src/web/handlers/GestureHandler.ts [hunk 4]
```diff
@@ -394,62 +397,93 @@ export default abstract class GestureHandler {
...
-        absoluteX: event.changedTouches[i].clientX,
-        absoluteY: event.changedTouches[i].clientY,
-      });
+    let numberOfTouches: number = all.length;
+
+    if (
+      event.eventType === EventTypes.UP ||
+      event.eventType === EventTypes.ADDITIONAL_POINTER_UP
+    ) {
+      --numberOfTouches;
```
Comment (j-piasecki) [inline src/web/handlers/GestureHandler.ts:476]: I assume this is to be compatible with native implementation, if so I'd mention it in comment

File: example/src/new_api/manualGestures/index.tsx [hunk 5]
```diff
@@ -82,7 +82,7 @@ export default function Example() {
       }
     })
     .onTouchesCancelled((e, manager) => {
-      for (const touch of e.changedTouches) {
+      for (const touch of e.allTouches) {
```
Comment (j-piasecki) [inline example/src/new_api/manualGestures/index.tsx:85]: In case of cancel, wouldn't both arrays contain exactly the same pointers?

**Review by j-piasecki — CHANGES_REQUESTED** [review]

File: example/src/new_api/manualGestures/index.tsx (hunk 5 above)
Comment (Warus15) [reply example/src/new_api/manualGestures/index.tsx:85]: At first they didn't and that's why I changed that. Then I slightly changed cancel behavior and right now they do contain both. However I left that change to emphasize that we do cancel every pointer

**Review by j-piasecki — APPROVED** [review]

**Pull request merged**